### PR TITLE
Vishala Consumables Page View

### PR DESCRIPTION
**Pull request by Vishala09**
# Description
This PR is about the Consumables List View from Phase 2 with Sorting based on project name and item name , Filtering based on selected values of project and consumable  and Modal implementation for viewing update and purchase history.
Implements # (WBS)  8.1.1

## Related PRS (if any):
This frontend PR is related to the [#672](https://github.com/OneCommunityGlobal/HGNRest/pull/672) backend PR.

## Main changes explained:
- Created components for Consumables

## How to test:
1. check into current branch
2. do `npm install` and `npm run start:local` to run this PR locally
3. Clear site data/cache
4. log as admin user
5. verify all the table functionalities like open/close modals , sorting and filtering

## Screenshots or videos of changes:

[Untitled_ Dec 29,2023 5_14 PM.webm](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/65274029/b86e7b96-4b0a-4572-a4c9-10a059a5400b)

### CODE REVIEW

**Review by abdel-lall — DISMISSED** [review]
Works as described. Good job!

Screenshot 2023-12-29 234022

**Review by Alforoan — DISMISSED** [review]
All functionalities seem to work as intended. Nice!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/118868270/f540901e-6bac-4ee8-a1a3-602781285d2a

**Review by scoopstroop92 — COMMENTED** [review]
Hello @Vishala09  Good job on the functionality. I just notice one issue. First time I  load the page  the table is empty  After I make a selection/interaction in the UI then I see the list of projects in the table. Let me know if this is an acceptable behaviour

I repeated this couple of times to make sure if its a repeated behaviour. After clearing the cache when the page is loaded for the first time this happens
 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/89441435/1182cdb4-1a34-42e4-a5a9-cd9de693a478

**Review by bienzguoa — COMMENTED** [review]
Same case as [ramyaram2092] mentioned above, after refreshing, it has projects displayed. Functionality works as intended.
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/92126559/b2d1cf88-585b-4090-a65e-c93f7d11b09d

File: src/components/BMDashboard/Consumables/ConsumablesList/Consumables.css
Comment (ChengyanOo) [inline src/components/BMDashboard/Consumables/ConsumablesList/Consumables.css:1]: Hi @Vishala09, 

I saw that you are using px and rem here, would it be better if we consider using the same size unit?

**Review by ChengyanOo — DISMISSED** [review]
Hey @Vishala09,

Just went through your code updates – they're super neat and clean, nicely done! Tested out the new feature you implemented as well, and it’s working smoothly. I’ve included a video below for a more detailed walkthrough. Awesome work on the implementation, everything's spot on as you described!


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/69735000/b7230ac1-5c40-4331-be64-426f1a8bade0

**Comment by Vishala09** [comment]
> Hello @Vishala09 Good job on the functionality. I just notice one issue. First time I load the page the table is empty After I make a selection/interaction in the UI then I see the list of projects in the table. Let me know if this is an acceptable behaviour
> I repeated this couple of times to make sure if its a repeated behaviour. After clearing the cache when the page is loaded for the first time this happens

This issue is resolved. Please do a git pull and re test if necessary.

**Comment by Vishala09** [comment]
> Same case as [ramyaram2092] mentioned above, after refreshing, it has projects displayed. Functionality works as intended. https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/92126559/b2d1cf88-585b-4090-a65e-c93f7d11b09d

Thank you for testing. This issue is resolved. Please do a git pull and test if needed.

**Review by tdkent — DISMISSED** [review]

**Review by scoopstroop92 — DISMISSED** [review]
The issue is fixed. I dont see it anymore . Good job Vishala !
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/89441435/1d5471d4-3d2e-403d-abcb-24c8549f35e9

**Review by tdkent — APPROVED** [review]

**Pull request merged**